### ISSUE TITLE
Variable sensor sampling rates

### ISSUE THREAD

**Issue by mciantyre** [ebelski/rust-copter#44]
Our 0.1 demo queries all readings (accelerometer, gryoscope, magnetometer) from an MPU9250 at 100Hz. This happens

- regardless of the sensor sampling rate, which might be less than 100Hz
- even if we don't care about that sensor (i.e. we're not using the magnetometer)

There's no reason we're doing this, other than it was simple to put together.

A future embedded platform should support variable sensor sampling rates. This will let us express sensor reading strategies like "poll acc at 2KHz, gyro at 250Hz, and mag at 4Hz."

We should be able to change sampling rates during runtime, before takeoff. Supporting sampling rate changes during flight are not supported. Let me know if changing sensor sampling rates during flight might be useful. I'd need some help thinking through the details.

**Comment by ebelski** [comment]
I don't think we need to vary sampling rates during flight. That'd be scary actually 😨 we'd probably get a situation like this 

```
A 747 was flying along and was full of Polish people. 
As they were going past some beautiful landmarks, the pilot came over 
the intercom and instructed all who were interested in seeing the 
landmark to look out the right side of the plane. Many 
passengers did so, and the plane promply crashed. Why?
Too many poles in the right hand plane.
```
